src/plugins/intel_npu/src/al/src/sync_infer_request.cpp

```diff
@@ -150,19 +151,23 @@ void SyncInferRequest::check_tensor(const ov::Output<const ov::Node>& port,
 void SyncInferRequest::check_tensors() const {
     const auto& inputs = _compiledModel->inputs();
     for (size_t i = 0; i < inputs.size(); i++) {
-        check_tensor(inputs[i], _allTensors.at(inputs[i].get_node()->get_friendly_name()));
+        if (_allTensors.find(inputs[i].get_node()->get_friendly_name()) != _allTensors.end()) {
```

Comment:
Actually not really. Not all of them should be already allocated at this stage. Only if calling set_tensor or get_tensor a new tensor will be allocated and will be available here. Otherwise it will be created when infer first time.

Reply:
Yeah, I wanted just to remove it. But I said to keep it to check at least the tensors that are already created. But yes, you are true. I had this remark as well before starting to implement it.

Checking `_copyAllTensors `would mean the same thing, more than that, sometimes we can have only the user tensor allocated at this stage and not the L0 tensor, so it could check fewer tensors at all.

In my opinion make more sense to check `_allTensors` here.